app/src/main/java/org/wikipedia/analytics/eventplatform/ArticleInteractionEvent.kt

```diff
@@ -0,0 +1,154 @@
+package org.wikipedia.analytics.eventplatform
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.wikipedia.analytics.eventplatform.ArticleInteractionEvent.ActionType.*
+import org.wikipedia.auth.AccountUtil
+
+class ArticleInteractionEvent(private var wikiDb: String, private var pageId: Int) : TimedEvent() {
+
+    private lateinit var action: String
```

Comment:
We no longer need this field here. We can just make `submitEvent(...)` take an action as a parameter.

And does the action really need to be an Enum? (Can it just be literal string values? Are they used anywhere else?)